utils/nbd-log.h

```diff
@@ -28,59 +28,48 @@
 
 #include "nbd-sysconfig.h"
 
-static inline int _nbd_err(const char *fmt, ...)
-{
-    va_list ap;
-    int ret = 0;
-
-    va_start(ap, fmt);
-
-    ret = vfprintf(stderr, fmt, ap);
-    va_end(ap);
-
-    return ret;
-}
-
-static inline int _nbd_out(const char *fmt, ...)
-{
-    va_list ap;
-    int ret = 0;
-
-    va_start(ap, fmt);
-
-    ret = vprintf(fmt, ap);
-    va_end(ap);
-
-    return ret;
-}
-
-#define nbd_err(...)                                                \
-    do {                                                            \
-        _nbd_err(__VA_ARGS__);                                      \
-        syslog(LOG_ERR, __VA_ARGS__);                               \
-    } while (0)
-
-#define nbd_out(...)                                                \
-    do {                                                            \
-        _nbd_out(__VA_ARGS__);                                      \
-        syslog(LOG_INFO, __VA_ARGS__);                              \
-    } while (0)
-
-#define nbd_dbg(...)                                                \
-    do {                                                            \
-            _nbd_out(__VA_ARGS__);                                  \
-            syslog(LOG_DEBUG, __VA_ARGS__);                         \
-    } while (0)
-
-static inline int nbd_log_init(void)
-{
-    openlog(NULL, 0, 0);
-    return 0;
-}
-
-static inline void nbd_log_destroy(void)
-{
-    closelog();
-}
+#define NBD_LOG_CRIT	LOG_CRIT	/* critical conditions */
+#define NBD_LOG_ERROR	LOG_ERR		/* error conditions */
+#define NBD_LOG_WARN	LOG_WARNING	/* warning conditions */
+#define NBD_LOG_INFO	LOG_INFO	/* informational */
+#define NBD_LOG_DEBUG	LOG_DEBUG	/* debug-level messages */
+#define NBD_LOG_DEBUG_IO   (LOG_DEBUG + 1)	/* nbd io messages */
+
+struct nbd_device;
+
+void nbd_set_log_level(int level);
+unsigned int nbd_get_log_level(void);
+int nbd_setup_log(char *log_dir);
+void nbd_destroy_log(void);
+int nbd_make_absolute_logfile(char *path, const char *filename);
```

Comment:
Please remove this and make it as static in the nbd-log.c file.

Reply:
OK